File: python code/binary image processing/dilation.py
import cv2

# read image
image = cv2.imread("../../images/truth.png", cv2.IMREAD_COLOR)

# check for invalid input
if image is None:
    print("Could not open or find image")

# get structuring kernel which used for in dilation
dilationSize = 6
element = cv2.getStructuringElement(shape=cv2.MORPH_CROSS,
                                    ksize=(2 * dilationSize + 1, 2 * dilationSize + 1),
                                    anchor=(dilationSize, dilationSize))


# dilating the image
dilatedImage = cv2.dilate(src=image, kernel=element)

# display images
cv2.imshow("Original Image", image)
cv2.imshow("Dilated Image", dilatedImage)

cv2.waitKey(0)

File: c++ code/week1/examples/binary image processing/binary image processing/dilation.cpp
#include <opencv2/opencv.hpp>

#include <iostream>

int main2() {
	// read image in and store in cv::Mat image
	cv::Mat image = cv::imread("../../../../../images/truth.png", cv::IMREAD_COLOR);

	// create a structuring element
	int dilationSize = 6;
	cv::Mat element = cv::getStructuringElement(cv::MORPH_CROSS,
		cv::Size(2 * dilationSize + 1, 2 * dilationSize + 1),
		cv::Point(dilationSize, dilationSize));

	// dilated image stored in cv::Mat
	cv::Mat dilatedImage;

	// dilation will increase brightness
	cv::dilate(image, dilatedImage, element);

	// display images
	cv::imshow("Original Image", image);
	cv::imshow("Dilated Image", dilatedImage);

	cv::waitKey(0);

	return EXIT_SUCCESS;
}